### PR TITLE
Add Knapsack0-1.py

### PR DESCRIPTION
**Pull request by kartikp36**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes
### Description

Knapsack 0-1 algorithm in python using dynamic programming.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Second part of issue #1440
First Part: #1537

### CODE REVIEW

File: Python/dp/Knapsack0-1.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    # knapsack value is it's value + best we can do(ie. value of knapsack_table):
+                    # 1) without this item(with the previous one) and
+                    # 2) capacity j - weight of ith item
+                    values[i-1] + knapsack_table[i-1][j-weights[i-1]],
+                    knapsack_table[i-1][j])  # without ith item
+
+    return knapsack_table[items][capacity]
+
+
+print("Enter Capacity:")
```
Comment (HarshCasper) [inline Python/dp/Knapsack0-1.py]: Enclose this in a `if __name__ == '__main__'`

File: Python/dp/Knapsack0-1.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+W: Capacity of knapsack
+Time Complexity: O(N*W).
+Space Complexity: O(N*W). 
+
+
+"""
+
+
+def knapsack(capacity, weights, values):
+    # items= number of items
```
Comment (HarshCasper) [inline Python/dp/Knapsack0-1.py]: Kindly make the comments in full sentences

File: Python/dp/Knapsack0-1.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Knapsack 0-1 problem using dp (0-1 means we either choose it or we don't, no fractions)
+Given weights and their corresponding values,
+We fill knapsack of capacity W to obtain maximum possible value in bottom-up manner.
+
+N: Number of (items)weight elements 
+W: Capacity of knapsack
+Time Complexity: O(N*W).
+Space Complexity: O(N*W). 
+
+
```
Comment (HarshCasper) [inline Python/dp/Knapsack0-1.py]: Trim extra lines!

File: Python/dp/Knapsack0-1.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            # weight of item is greater than current capacity, so we cannot use it
+            # so knapsack value here is the best we can do without this item ie. with the (i-1)th item
+            elif weights[i-1] > j:
+                knapsack_table[i][j] = knapsack_table[i-1][j]
+
+            # weight of item is less than capacity, so can use it to optimize knapsack
+            else:
+                knapsack_table[i][j] = max(
+                    # if ith item is included,
```
Comment (HarshCasper) [inline Python/dp/Knapsack0-1.py]: Can you try making the Comments more consistent as per grammar (That's not a necessity but it would look good 😅 )

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by kartikp36** [comment]
Sorry, I made the changes to the file, but then accidentally merged the latest NeoAlgo/master to this branch instead of my fork's master. Should i create a fresh PR ?

**Comment by kartikp36** [comment]
@HarshCasper Please review my changes and the mistake.

**Pull request closed without merging**